### PR TITLE
Take into accounts tokens on the java package declaration, fixes #23

### LINKED ISSUE #23: Imports are wrongly flagged as unused and removed
This is a bit of a corner case, but the input file is [here](https://github.com/apache/camel/blob/master/components/camel-jaxb/src/test/java/org/apache/camel/example/package-info.java) and the 2 imports are removed.

```
@XmlSchema(
    xmlns = { 
        @XmlNs(prefix = "order", namespaceURI = "http://www.camel.apache.org/jaxb/example/order/1"),
        @XmlNs(prefix = "address", namespaceURI = "http://www.camel.apache.org/jaxb/example/address/1")
        }
)
package org.apache.camel.example;

import javax.xml.bind.annotation.XmlNs;
import javax.xml.bind.annotation.XmlSchema;
```

### PR DESCRIPTION
**Pull request by gnodet**

### CODE REVIEW

File: src/test/resources/UnusedImports2.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@XmlSchema(
+		xmlns = {
+				@XmlNs(prefix = "order", namespaceURI = "http://www.camel.apache.org/jaxb/example/order/1"),
+				@XmlNs(prefix = "address", namespaceURI = "http://www.camel.apache.org/jaxb/example/address/1")
+		}
+)
+package org.apache.camel.example;
+
+import javax.xml.bind.annotation.XmlNs;
+import javax.xml.bind.annotation.XmlSchema;
```
Comment (ctubbsii) [inline src/test/resources/UnusedImports2.java]: Can this test case be added to the existing unused imports test case, to keep things simpler?

File: src/test/java/net/revelc/code/impsort/ImpSortTest.java [hunk 2]
```diff
@@ -120,6 +120,17 @@ public void testUnused() throws IOException {
...
+  public void testUnused2() throws IOException {
+    Path p = Paths.get(System.getProperty("user.dir"), "src", "test", "resources",
+        "UnusedImports2.java");
+    Result result = new ImpSort(StandardCharsets.UTF_8, eclipseDefaults, true, true).parseFile(p);
+    Set<String> imports =
+        result.getImports().stream().map(Import::getImport).collect(Collectors.toSet());
+    assertEquals(2, imports.size());
+    assertEquals(2, result.getImports().size());
+  }
+
```
Comment (ctubbsii) [inline src/test/java/net/revelc/code/impsort/ImpSortTest.java]: Should combine this with the existing unused import test case, and verify that the contained imports are what is expected, rather than merely just check the count.

File: src/main/java/net/revelc/code/impsort/ImpSort.java [hunk 3]
```diff
@@ -216,12 +216,17 @@ private static void convertAndAddImport(LinkedHashSet<Import> allImports, List<N
...
-            .map(Optional::get).filter(r -> r != TokenRange.INVALID).flatMap(r -> {
-              // get all JavaTokens as strings from each range
-              return StreamSupport.stream(r.spliterator(), false);
-            }).map(JavaToken::asString);
+    Stream<Node> packageDecl =
+        unit.getPackageDeclaration().isPresent()
+            ? Stream.of(unit.getPackageDeclaration().get()).map(PackageDeclaration::getAnnotations)
+                .flatMap(NodeList::stream)
+            : Stream.empty();
+    Stream<String> typesInCode = Stream.concat(packageDecl, unit.getTypes().stream())
```
Comment (ctubbsii) [inline src/main/java/net/revelc/code/impsort/ImpSort.java]: Not sure how I feel about concat here, since this gets us closer to the code we had before it was recently cleaned up, when it was much harder to read (described at nightmarish in 6a760b37c4e4cd251cf5a811083844963f5ac7da).

If you have any ideas here for alternate implementations that could make this slightly more readable, I'm open to that. Otherwise, I'll defer any readability cleanup to a later task.

File: src/main/java/net/revelc/code/impsort/ImpSort.java [hunk 4]
```diff
@@ -216,12 +216,17 @@ private static void convertAndAddImport(LinkedHashSet<Import> allImports, List<N
...
-        unit.getTypes().stream().map(TypeDeclaration::getTokenRange).filter(Optional::isPresent)
-            .map(Optional::get).filter(r -> r != TokenRange.INVALID).flatMap(r -> {
-              // get all JavaTokens as strings from each range
-              return StreamSupport.stream(r.spliterator(), false);
-            }).map(JavaToken::asString);
+    Stream<Node> packageDecl =
+        unit.getPackageDeclaration().isPresent()
+            ? Stream.of(unit.getPackageDeclaration().get()).map(PackageDeclaration::getAnnotations)
+                .flatMap(NodeList::stream)
+            : Stream.empty();
```
Comment (ctubbsii) [inline src/main/java/net/revelc/code/impsort/ImpSort.java]: This becomes much cleaner on Java9 and later, with `Optional::stream`, but we're not there yet. :smiley_cat:

File: src/test/resources/UnusedImports2.java (hunk 1 above)
Comment (ctubbsii) [reply src/test/resources/UnusedImports2.java:23]: I did this myself when I merged.

File: src/test/java/net/revelc/code/impsort/ImpSortTest.java (hunk 2 above)
Comment (ctubbsii) [reply src/test/java/net/revelc/code/impsort/ImpSortTest.java:133]: I did this myself when I merged.
